app/src/main/java/org/mozilla/fenix/home/sessioncontrol/viewholders/pocket/PocketStoriesComposables.kt

```diff
@@ -117,23 +127,175 @@ fun PocketStories(
 
     LazyRow {
         itemsIndexed(stories) { index, item ->
-            if (index < halfStoriesIndex) {
-                Column(
-                    Modifier.padding(end = if (index == halfStoriesIndex) 0.dp else 8.dp)
-                ) {
-                    PocketStory(item)
+            key(item.title) {
+                if (index < halfStoriesIndex) {
+                    Column(
+                        Modifier.padding(end = if (index == halfStoriesIndex) 0.dp else 8.dp)
+                    ) {
+                        PocketStory(item)
 
-                    Spacer(modifier = Modifier.height(24.dp))
+                        Spacer(modifier = Modifier.height(24.dp))
 
-                    stories.getOrNull(halfStoriesIndex + index)?.let {
-                        PocketStory(it)
+                        stories.getOrNull(halfStoriesIndex + index)?.let {
+                            PocketStory(it)
+                        }
                     }
                 }
             }
         }
     }
 }
 
+/**
+ * Displays a list of [PocketRecommendedStoryCategory].
+ *
+ * @param categories the categories needed to be displayed.
+ * @param onCategoryClick callback for when the user taps a category.
+ */
+@Composable
+fun PocketStoriesCategories(
+    categories: List<PocketRecommendedStoryCategory>,
+    onCategoryClick: (PocketRecommendedStoryCategory) -> Unit
+) {
+    StaggeredHorizontalGrid {
+        categories.forEach { category ->
+            PocketStoryCategory(category) {
+                onCategoryClick(category)
+            }
+        }
+    }
+}
+
+/**
+ * Displays an individual [PocketRecommendedStoryCategory].
+ *
+ * @param category the categories needed to be displayed.
+ * @param onClick callback for when the user taps this category.
+ */
+@Composable
+fun PocketStoryCategory(
+    category: PocketRecommendedStoryCategory,
+    onClick: (Boolean) -> Unit
+) {
+    val contentColor = when (category.isSelected) {
+        true -> Color.Blue
+        false -> Color.DarkGray
+    }
+
+    OutlinedButton(
+        onClick = { onClick(category.isSelected) },
+        shape = RoundedCornerShape(32.dp),
+        border = BorderStroke(1.dp, contentColor),
+        colors = ButtonDefaults.outlinedButtonColors(
+            contentColor = contentColor
+        ),
+        contentPadding = PaddingValues(8.dp, 7.dp)
+    ) {
+        Row {
+            Text(
+                text = category.name,
+                modifier = Modifier.alignByBaseline(),
+            )
+            Icon(
+                painter = painterResource(id = R.drawable.mozac_ic_check),
+                contentDescription = "Expand or collapse Pocket recommended stories",
+                modifier = Modifier.alignByBaseline()
```

Comment:
The baselines still don't seem to be aligned. Not sure about the final design so I didn't worry much about it. If needed we'll be able to quickly spin up a custom view for this.

Reply:
In the final design this icon won't be needed. So not a priority to work here perfectly.